applications/GeoMechanicsApplication/tests/point_load_semi_analytical_solution.py

```diff
@@ -0,0 +1,230 @@
+
+    Attributes:
+        - young (float): Young's modulus [Pa]
+        - poisson (float): Poisson's ratio [-]
+        - density (float): density [kg/m^3]
+        - load (float): load value [N/m^2]
+        - integral_stepsize (float): dimensionless time step size for solving integrals (default = 0.001) [-]
+        - shear_modulus (float): shear modulus [Pa]
+        - p_wave_modulus (float): P-wave modulus [Pa]
+        - cs (float): shear wave velocity [m/s]
+        - cp (float): compression wave velocity [m/s]
+        - eta (float): ratio of shear wave to compression wave velocity [-]
+        - epsilon (float): epsilon to avoid division by zero [-]
+
+    """
+
+    def __init__(self,
+                 youngs_modulus: float,
+                 poisson_ratio: float,
+                 density: float,
+                 load: float,
+                 integral_stepsize: float = 0.001):
+        """
+        Constructor for the point load class
+
+        Args:
+            - youngs_modulus (float): Young's modulus [Pa]
+            - poisson_ratio (float): Poisson's ratio [-]
+            - density (float): density [kg/m^3]
+            - load (float): load value [N]
+            - integral_stepsize (float): dimensionless time step size for solving integrals (default = 0.001) [-]
+        """
+
+        self.young = youngs_modulus
+        self.poisson = poisson_ratio
+        self.density = density
+        self.load = load
+
+        # dimensionless time step size for solving integrals
+        self.integral_stepsize = integral_stepsize
+
+        # calculate derived elastic properties
+        self.shear_modulus = self.young / (2 * (1 + self.poisson))
+        self.p_wave_modulus = self.young * (1 - self.poisson) / ((1 + self.poisson) * (1 - 2 * self.poisson))
+        self.cs = math.sqrt(self.shear_modulus / self.density)  # shear wave velocity
+        self.cp = math.sqrt(self.p_wave_modulus / self.density)  # compression wave velocity
+
+        self.eta = self.cs / self.cp  # ratio of shear wave to compression wave velocity
+
+        # epsilon to avoid division by zero
+        self.epsilon = self.integral_stepsize**2
+
+
+    def __wpekeris(self, nu: float, t: float) -> float:
+        pi = 4.0 * math.atan(1.0)
+        fac = 1.0 / (2.0 * pi * pi)
+        eps = 0.000001
+        eps *= eps
+        eps1 = 0.001
+        nn = (1.0 - 2.0 * nu) / (2.0 * (1.0 - nu))
+        n = math.sqrt(nn)
+        e = 0.000001
+        e *= e
+        f = 1.0
+        b = (1.0 - nu) / 8.0
+
+        if nu > 0.1:
+            while f > e:
+                a = b
+                b = (1.0 - nu) / (8.0 * (1.0 + a) * (nu + a))
+                f = math.fabs(b - a)
+        else:
+            while f > e:
+                a = b
+                b = math.sqrt((1.0 - nu) / (8.0 * (1.0 + a) * (1.0 + nu / a)))
+                f = math.fabs(b - a)
+
+        tr = math.sqrt(1.0 + b)
+        if t <= n:
+            g = 0
```

Comment:
The returned result is of type `float`, so I would use `0.0` rather than `0`:
```suggestion
            g = 0.0
```